src/NUnitFramework/framework/Constraints/ExactCountConstraint.cs

```diff
@@ -31,9 +31,19 @@ namespace NUnit.Framework.Constraints
     /// item in a collection, succeeding only if a specified
     /// number of items succeed.
     /// </summary>
-    public class ExactCountConstraint : PrefixConstraint
+    public class ExactCountConstraint : Constraint
```

Comment:
Maybe we really want two different constraints, analogous to PropertyConstraint and PropertyExistsConstraint. Of course, you have it working already, but it seems a little awkward to have one constraint that does two things.

In the case of `Has.Property` the prop operator can end up creating either a of the two constraints, depending on whether or not it comes at the end of the expression. Perhaps Items could do the same thing.